src/placard.cpp

```diff
@@ -0,0 +1,212 @@
+#include "placard.hpp"
+
+Texture Placard::placard_texture;
+
+Placard::Placard(vec2 parent_position, float parent_rotation){
+  if (!placard_texture.is_valid()) {
+    if (!placard_texture.load_from_file(textures_path("SignSheetv2.png"))) {
+      fprintf(stderr, "Failed to load sign texture!\n");
+      //return false;
+    }
+  }
+
+  // The position (0,0) corresponds to the center of the texture
+  m_wr = placard_texture.width * 0.5;
+  m_hr = placard_texture.height * 0.5;
+
+  m_sprite_width = placard_texture.width / 4;
+
+  //initialize the map telling which section of the spritesheet to show
+  m_texture_coords[turn_direction::STRAIGHT] = {0.f, 0.25f};
+  m_texture_coords[turn_direction::RIGHT] = {0.25f, 0.5f};
+  m_texture_coords[turn_direction::LEFT] = {0.5f, 0.75f};
+  m_texture_coords[turn_direction::EMERGENCY] = {0.75f, 1.f};
+
+
+  //for setting the array below: x refers to the left boundary of the sprite,
+  //y refers to the right boundary (NOT vertical positioning)
+  //sprites are laid out in a single row in the spritesheet, so they all use the full height
+	TexturedVertex vertices[4];
+	vertices[0].position = { -m_wr, +m_hr, 0.f };
+	vertices[0].texcoord = { m_texture_coords[m_desired_turn].x, 1.f };//top left
+	vertices[1].position = { -m_wr + m_sprite_width, +m_hr, 0.f };
+	vertices[1].texcoord = { m_texture_coords[m_desired_turn].y, 1.f };//top right
+	vertices[2].position = { -m_wr + m_sprite_width, -m_hr, 0.f };
+	vertices[2].texcoord = { m_texture_coords[m_desired_turn].y, 0.f };//bottom right
+	vertices[3].position = { -m_wr, -m_hr, 0.f };
+	vertices[3].texcoord = { m_texture_coords[m_desired_turn].x, 0.f };//bottom left
+
+	// counterclockwise as it's the default opengl front winding direction
+	uint16_t indices[] = { 0, 3, 1, 1, 3, 2 };
+
+	// Clearing errors
+	gl_flush_errors();
+
+	// Vertex Buffer creation
+	glGenBuffers(1, &mesh.vbo);
+	glBindBuffer(GL_ARRAY_BUFFER, mesh.vbo);
+	glBufferData(GL_ARRAY_BUFFER, sizeof(TexturedVertex) * 4, vertices, GL_STATIC_DRAW);
+
+	// Index Buffer creation
+	glGenBuffers(1, &mesh.ibo);
+	glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, mesh.ibo);
+	glBufferData(GL_ELEMENT_ARRAY_BUFFER, sizeof(uint16_t) * 6, indices, GL_STATIC_DRAW);
+
+	// Vertex Array (Container for Vertex + Index buffer)
+	glGenVertexArrays(1, &mesh.vao);
+	if (gl_has_errors())
+    fprintf(stderr, "Failed to initialize placard graphics!\n");
+		//return false;
+
+	// Loading shaders
+	if (!effect.load_from_file(shader_path("textured.vs.glsl"), shader_path("textured.fs.glsl")))
+    fprintf(stderr, "Failed to load shaders for signs!\n");
+		//return false;
+
+  m_scale = {1.f, 1.f};
+  m_rotation = parent_rotation + M_PI / 2;
```

Comment:
M_Pi is an undeclared identifier for me, which causes a build error. in other files we just defined PI as a static variable at the top, are you using a math package or something?